dnscan.py

```diff
@@ -303,12 +303,25 @@ def zone_transfer(domain, ns):
         pass
 
 def add_target(domain):
-    if '%%' in domain:
-        for word in wordlist:
-            queue.put(domain.replace(r'%%', word))
-    else:
-        for word in wordlist:
-            queue.put(word + "." + domain)
+    for word in wordlist:
+        patterns = [word]
+        if args.alt:
+            probes = ["dev", "prod", "stg", "qa", "uat", "api", "alpha", "beta",
+                      "cms", "test", "internal", "staging", "origin", "stage"]
+            for probe in probes:
+                if probe not in word: # Reduce alterations that most likely don't exist (e.i. dev-dev.domain.com)
+                    patterns.append(probe + word)
+                    patterns.append(word + probe)
+                    patterns.append(probe + "-" + word)
+                    patterns.append(word + "-" + probe)
+            if not word[-1].isdigit(): # Limit the number suffix from 1-9
+                for n in range(1, 10):
```

Comment:
Do you think it would be useful to prefix these with a zero? I feel like I see stuff like `web01` more often than `web1`?

Reply:
Nah! I have been running a lot of subdomains enumeration so far but see `dev1`, `prod2` more likely than `dev01`, ...

Reply:
I will check back my database in order to confirm the appearance of `01`, `02`, ... then will come to a conclusion (if there are a lot of `01`, ...)

Reply:
Okay so there are, but there aren't much, so I don't think I will do it

Reply:
That's interesting - perhaps it's more of a thing people do on internal networks (and actual hostnames), rather than the public DNS records.

Do you often see things like `dev8` or `dev9`? I wonder if something like 0-5 and 01-05 would find more hosts (in the same amount of time)?